tests/test_opt.py

```diff
@@ -22,6 +25,39 @@ def test_naive_bcast_rv_lift():
     assert res is X_rv
 
 
+def test_naive_bcast_rv_lift_valued_var():
+    r"""Check that `naive_bcast_rv_lift` handles valued variables correctly."""
+
+    x_rv = at.random.normal(name="x")
+
+    mu = at.broadcast_to(x_rv, (2,))
+    y_rv = at.random.normal(mu, name="y")
+
+    x_vv = x_rv.clone()
+    y_vv = y_rv.clone()
+    logp_map = factorized_joint_logprob({x_rv: x_vv, y_rv: y_vv})
+
+    assert x_vv in logp_map
+    assert y_vv in logp_map
+    y_val = np.array([0, 0])
+    assert np.allclose(
+        logp_map[y_vv].eval({x_vv: 0, y_vv: y_val}), st.norm(0).logpdf(y_val)
+    )
+
+    # Lifting should also work when `BroadcastTo`s are directly assigned value
```

Comment:
I don't think this example should be valid. We are measuring twice the same thing (pre-broadcast and post broadcasted value variable)

Besides, this would be fine with saying `{x_vv: 0, z_vv: [1, 1]}` which does not match the original graph

Reply:
If either only one of the pre- and post- broadcasted variable is valued then #116 already worked.

... although the post-broadcast case is still a bit iffy and depends on what we mean by joint-logprob. The original generative graph of `z_rv` would correspond to a logp of something like:

```
def broadcastedTo_logp(value, original_shape, dist_op, dist_params):

  # e.g., undo_broadcast((3, 3, 3), (1,)) -> (3,), True
  pre_broadcast_value, possible = undo_broadcast(value, original_shape)
  
  return switch(
      possible,
      logprob(dist_op, pre_broadcast_value, *dist_params),
      -inf,
      )
  )
```

Reply:
> If either only one of the pre- and post- broadcasted variable is valued then #116 already worked.

Only if it broadcasted something that wasn't valued, but that's the limitation I've demonstrated. 

> ... although the post-broadcast case is still a bit iffy and depends on what we mean by joint-logprob. Strictly speaking, the original generative graph of `z_rv` would correspond to a logp of something like:
> 
> ```python
> def broadcastTo_logp(value, original_shape, dist_op, dist_params):
> 
>   # e.g., undo_broadcast((3, 3, 3), (1,)) -> (3,)
>   pre_broadcast_value = undo_broadcast(value, original_shape)
>   
>   return switch(
>     at.eq(value, at.broadcast_to(pre_broadcast_value, value.shape)),
>     logprob(dist_op, pre_broadcast_value, *dist_params),
>     -inf,
>   )  
> ```
> 
> But that's a separate question from the comment above

You're goint to need to clarify this, because, right now, I don't see the relevance of this function.  Can you demonstrate the underlying issue by deriving an incorrect log-probability calculation under these changes?